examples/webusb/index.js

```diff
@@ -0,0 +1,290 @@
+    assert(!this.selected, 'Other device already in use.');
+    assert(this.deviceInfos.has(info), 'Could not find device.');
+
+    this.selected = info;
+    this.device = new Device({
+      device: info,
+      timeout: timeout
+    });
+
+    try {
+      await this.device.open();
+    } catch (e) {
+      console.error(e);
+      this.selected = null;
+      this.device = null;
+    }
+
+    return this.device;
+  }
+
+  async closeDevice(info) {
+    assert(this.selected, 'No device in use.');
+    assert(this.deviceInfos.has(info), 'Could not find device.');
+    assert(this.selected === info, 'Can not close closed device.');
+
+    await this.device.close();
+
+    console.log('device closed');
+
+    this.selected = null;
+    this.device = null;
+  }
+}
+
+const manager = new DeviceManager();
+const chooseBtn = document.getElementById('choose');
+const chosenDiv = document.getElementById('chosen');
+const devicesDiv = document.getElementById('devices');
+
+chooseBtn.addEventListener('click', async (event) => {
+  const device = await manager.requestDevice();
+
+  await manager.openDevice(device);
+
+  renderManager();
+});
+
+global.addEventListener('load', async () => {
+  await manager.open();
+
+  renderManager();
+});
+
+// We rerender all the time..
+// Use framework or something.
+
+function renderManager() {
+  const selected = manager.selected;
+  const devices = manager.getDevices();
+
+  console.log('Rendering page.', manager.selected);
+
+  renderChosen(chosenDiv, manager, selected);
+  renderDevices(devicesDiv, manager, devices);
+}
+
+function renderDevices(element, manager, devices) {
+  removeChildren(element);
+
+  for (const device of devices) {
+    renderDevice(element, manager, device);
+  }
+}
+
+function renderDevice(element, manager, info) {
+  const container = document.createElement('div');
+  const name = document.createElement('span');
+  const choose = document.createElement('button');
+
+  choose.innerText = 'Open.';
+  name.innerText = deviceInfoMini(info);
+
+  // we don't clean up listeners.. too much headache
+  choose.addEventListener('click', async () => {
+    await manager.openDevice(info);
+
+    renderManager();
+  });
+
+  container.appendChild(name);
+  container.appendChild(choose);
+
+  element.appendChild(container);
+}
+
+// NOTE: I don't take care of the unlisted DOM elements.
+function renderChosen(element, manager, info) {
+  removeChildren(element);
+
+  if (!info)
+    return;
+
+  const closeBtn = document.createElement('button');
+
+  closeBtn.innerText = 'Close.';
+  closeBtn.addEventListener('click', async function close() {
+    await manager.closeDevice(info);
+
+    closeBtn.removeEventListener('click', close);
+
+    renderManager();
+  });
+
+  const pubkeyBtn = document.createElement('button');
+  pubkeyBtn.innerText = 'Get public key';
+  pubkeyBtn.addEventListener('click', async () => {
+    const device = manager.device;
+
+    if (!device) {
+      alert('WTF..');
```

Comment:
Maybe a better error message? haha

Reply:
lol, forgot to change it. I will clean up comments :+1: